src/gnatss/ops/utils.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, List
+
+import numba
+import numpy as np
+from nptyping import NDArray, Shape
+
+DEFAULT_VECTOR_NORM = np.array([2.0, 0.0, 0.0])
+
+
+# Numba utilities
+
+
+@numba.njit(cache=True)
+def calc_uv(input_vector: NDArray[Shape["3"], Any]) -> NDArray[Shape["3"], Any]:
+    """
+    Calculate unit vector for a 1-D input vector of size 3
+
+    Parameters
+    ----------
+    input_vector : (3,) ndarray
+        A 1-D input vector as numpy array
+
+    Returns
+    -------
+    (3,) ndarray
+        The resulting unit vector as numpy array
+
+    Raises
+    ------
+    ValueError
+        If the input vector is not a 1-D array
+    """
+
+    if input_vector.ndim != 1 or input_vector.shape != (3,):
+        raise ValueError("Unit vector calculation must be 1-D array of shape 3!")
```

Comment:
I suggest adding the actual shape in the error message, that will make debugging easier.